invenio_rdm_records/records/dumpers/pids.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2021 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""DataCite-based data model for Invenio."""
+
+
+from invenio_records.dumpers import ElasticsearchDumperExt
+
+
+class ExternalPIDsDumperExt(ElasticsearchDumperExt):
+    """Elasticsearch dumper extension for External PIDs support.
+
+    The JSON representation of the external PIDs is a dictionary which keys
+    are the identifier's scheme. However, that would turn into a dynamic
+    mapping in ES with `text` type attributes. However, we want to be able to
+    search by e.g. DOI in exact match terms. Since it contains dots, slashes,
+    etc. it must be of type `keyword`. Therefore, this dumper turns the dict
+    into a fixed mapping, dumping the key into the scheme attribute.
+    """
+
+    def dump(self, record, data):
+        """Dump the data."""
+        pids = data.get('pids', {})
+        if pids:
+            dumped_pids = []
+            for scheme, pid_attrs in pids.items():
+                dumped_pids.append({
+                    "scheme": scheme,
+                    **pid_attrs
+                })
+
+            data['pids'] = dumped_pids
+
+    def load(self, data, record_cls):
+        """Load the data."""
+        pids = data.get('pids', {})
+        if pids:
+            loaded_pids = {}
+            for pid_attrs in pids:
+                scheme = pid_attrs.pop('scheme')
+                loaded_pids[scheme] = {**pid_attrs}
```

Comment:
Silly question: can you simply do
```suggestion
                loaded_pids[scheme] = pid_attrs
```

Reply:
🤦 yes... I like complexity 😂